### PR TITLE
Update IoT hub service client init

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: e2e/test/helpers/TestDevice.cs [hunk 1]
```diff
@@ -37,6 +37,7 @@ public sealed class TestDevice : IDisposable
         private static readonly IIotHubServiceRetryPolicy s_getRetryPolicy = new HubServiceTestRetryPolicy(s_getRetryableStatusCodes);
 
         private X509Certificate2 _authCertificate;
+        private static readonly IotHubServiceClient _client = new(TestConfiguration.IotHub.ConnectionString);
```
Comment (drwill-ms) [inline e2e/test/helpers/TestDevice.cs:41]: Cache this instance so it doesn't get created for each device. Wasteful!

File: iothub/service/src/ClientApiVersionHelper.cs [hunk 2]
```diff
@@ -9,11 +9,11 @@ namespace Microsoft.Azure.Devices
     internal static class ClientApiVersionHelper
     {
         private const string ApiVersionQueryPrefix = "api-version=";
-        private const string ApiVersionDefault = "2021-04-12";
+        internal const string ApiVersionDefault = "2021-04-12";
```
Comment (drwill-ms) [inline iothub/service/src/ClientApiVersionHelper.cs:12]: Make internal so we can reference it.

File: iothub/service/src/IotHubServiceClient.cs [hunk 3]
```diff
@@ -42,6 +44,13 @@ protected IotHubServiceClient()
         {
         }
 
+        private IotHubServiceClient(IotHubServiceClientOptions options)
```
Comment (drwill-ms) [inline iothub/service/src/IotHubServiceClient.cs:47]: Consolidate common ctor steps to reduce DRY violation.

**Review by abhipsaMisra — APPROVED** [review]

**Review by brycewang-microsoft — APPROVED** [review]

**Review by schoims — APPROVED** [review]

**Pull request merged**